backend/LexBoxApi/Controllers/UserController.cs

```diff
@@ -99,18 +135,31 @@ public async Task<ActionResult<LexAuthUser>> AcceptEmailInvitation(RegisterAccou
         }
 
         var jwtUser = _loggedInContext.User;
-
-        var hasExistingUser = await _lexBoxDbContext.Users.FilterByEmailOrUsername(accountInput.Email).AnyAsync();
-        acceptActivity?.AddTag("app.email_available", !hasExistingUser);
-        if (hasExistingUser)
+        if (jwtUser.Email != accountInput.Email)
         {
-            ModelState.AddModelError<RegisterAccountInput>(r => r.Email, "email already in use");
+            // Changing email address in invite links is not allowed; this prevents someone from trying to reuse a JWT belonging to somebody else
+            ModelState.AddModelError<RegisterAccountInput>(r => r.Email, "email address mismatch in invitation link");
             return ValidationProblem(ModelState);
         }
 
-        var userEntity = CreateUserEntity(accountInput, jwtUser);
+        // We now allow multiple invitations to be accepted by the same account, so only create one if there isn't one already
+        var userEntity = await _lexBoxDbContext.Users
+            .Where(u => u.Email == accountInput.Email)
+            .Include(u => u.Projects)
+            .Include(u => u.Organizations)
+            .FirstOrDefaultAsync();
+        acceptActivity?.AddTag("app.email_available", userEntity is null);
+        if (userEntity is null)
+        {
+            userEntity = CreateUserEntity(accountInput, jwtUser);
+            _lexBoxDbContext.Users.Add(userEntity);
+        }
+        else
+        {
+            UpdateUserMemberships(jwtUser, userEntity);
```

Comment:
would like this to return an error instead of the user already exists

Reply:
Done in commit 0000c866.

Reply:
Heh, I think I now have the lowest-numbered commit hash in our repository. Four initial zeroes.